src/Microsoft.Azure.WebJobs.Extensions.EventHubs/Triggers/EventHubTriggerAttributeBindingProvider.cs

```diff
@@ -61,9 +61,10 @@ public Task<ITriggerBinding> TryCreateAsync(TriggerBindingProviderContext contex
 
             if (!string.IsNullOrWhiteSpace(attribute.Connection))
             {
-                _options.Value.AddReceiver(resolvedEventHubName, _nameResolver.Resolve(attribute.Connection));
+                var connectionString = _config.FindConnectionString(attribute.Connection);
+                _options.Value.AddReceiver(resolvedEventHubName, connectionString);
```

Comment:
The NameResolver does not check ConnectionStrings first. Since we're moving away from that for environment lookups, I'm going directly against IConfiguration now. As mentioned elsewhere, triggers do not go through the AttributeCloner, so they need to resolve these things themselves.